src/index.ts

```diff
@@ -5,10 +5,20 @@ import { JsonRPCRequest, JsonRPCResponse } from "web3/providers";
 
 const { fetch, Headers } = fetchPonyfill();
 
+const RATE_LIMIT_STATUS = 429;
+const DEFAULT_MAX_RETRIES = 3;
+const DEFAULT_RETRY_INTERVAL = 1000;
+const DEFAULT_RETRY_JITTER = 250;
+
 export interface AlchemyWeb3Config {
-  writeProvider?: Provider;
+  writeProvider?: Provider | null;
+  maxRetries?: number;
+  retryInterval?: number;
+  retryJitter?: number;
 }
 
+type FullConfig = { [K in keyof AlchemyWeb3Config]-?: AlchemyWeb3Config[K] };
```

Comment:
What does this syntax mean?

Reply:
@dtran320 This is a [mapped type](https://www.typescriptlang.org/docs/handbook/advanced-types.html). It defines `FullConfig` as the same type as `AlchemyWeb3Config`, except that none of its fields are optional. The way you read this is that `FullConfig` represents the type of an object whose keys are the same as those of `AlchemyWeb3Config`, and for each such key `K`, its value has the same type as the value for that key in `AlchemyWeb3Config` as indicated by the `AlchemyWeb3Config[K]`, but the "optional" annotation is removed as indicated by the `-?`.